src/main/java/cart/application/CartItemService.java

```diff
@@ -82,8 +78,19 @@ public void remove(Member member, Long id) {
     public CheckoutResponse checkout(Member member, List<Long> checkedCartItemIds) {
         List<CartItem> memberCartItems = cartItemRepository.findByMember(member);
         List<CartItem> checkedCartItems = calculateCheckedCartItems(checkedCartItemIds, memberCartItems);
-
-        return CheckoutResponse.of(checkedCartItems, member, pointDiscountPolicy, pointEarnPolicy);
+        Money totalPrice = calculateTotalPriceOfCheckedCartItems(checkedCartItems);
+
+        Money earnedPoints = PointEarnPolicy.DEFAULT.calculateEarnPoints(totalPrice);
+        Money moneyCondition = PointDiscountPolicy.DEFAULT.calculatePointCondition(totalPrice);
```

Comment:
리팩토링 잘 해주셨네요 👍🏻 

지금도 잘 해주셨지만 지토가 얼만큼 추상화할 것이냐에 따라 `DEFAULT` 정책을 드러내지 않을 수도 있을 것 같아요.

```suggestion
        Money earnedPoints = PointEarnPolicy.calculateEarnPoints(totalPrice);
        Money moneyCondition = PointDiscountPolicy.calculatePointCondition(totalPrice);
```

이렇게 되면 추후 새로운 정책이 추가되거나 조건에 따라 정책을 다르게 지정하더라도 변경되는 범위가 `calculateXX()` 함수로 한정될 수 있을 것 같네요.

Reply:
오 말씀해주신 부분은 떠올리지 못했네요!

말씀하신대로 `enum`에서 해당 메소드를 호출할 때, 메소드 내부에서 직접 해당 정책을 지정해주는 편이 더 좋을 것 같습니다!

제 경우 이를 해결하기 위해서 정책을 사용하는 행위를 인터페이스로 추상화하고, 구현체에서 `PolintDiscountPolicy.DEFAULT`를 사용하는 방법밖에는 떠오르지 않아서 이와 같이 고정해버리는 식으로 리펙토링을 진행했고, 정책이 변경될 때 마다 해당 정책을 사용하는 모든 코드가 영향을 받는다는게 좀 찝찝했었는데 왜 이런 방법을 떠올리지 못했나 싶네요..

정말 감사합니다!